markdown/developers/adding_pipelines.md

```diff
@@ -353,10 +362,21 @@ You will find the following files in each nf-core pipeline. They are automatical
 
 * `docs/`: Markdown files for documenting the pipeline
 
-* `lib/`: The lib directory contains groovy utility functions. Currently, this includes the `Schema.groovy`, `Headers.groovy`, `Completion.groovy` and `Checks.groovy` files. These are called from within the nf-core pipeline to do common pipeline tasks, like formatting the header for the command line. Any larger groovy function you want to call in the code should therefore go in here.
+* `lib/`: The lib directory contains Groovy utility functions. These are called from within the nf-core pipeline to do common pipeline tasks (e.g. parameter schema validation) and to hold Groovy functions that may be useful in the pipeline context (e.g. to validate pipeline-specific parameters). Currently, the following files are included:
+
+  * `NfcoreSchema.groovy`
+  * `NfcoreTemplate.groovy`
+  * `Utils.groovy`
+  * `WorkflowMain.groovy`
+  * `WorkflowPipeline.groovy`
+  * `nfcore_external_java_deps.jar`
```

Comment:
```suggestion
  * `NfcoreSchema.groovy` - Functions to validate input parameters using the pipeline JSON schema
  * `NfcoreTemplate.groovy` - Additional nf-core specific pipeline functions (sending emails, checking nf-core config profiles and more)
  * `Utils.groovy` - Additional generic pipeline functions (checking conda config and more)
  * `WorkflowMain.groovy` - Startup functions for the main pipeline (printing logs, custom params initialisation and more)
  * `WorkflowPipeline.groovy` - Functions for pipeline subworkflows
  * `nfcore_external_java_deps.jar` - Bundled Groovy dependencies so that pipelines work offline (mostly for JSON schema validation - see imports in `NfcoreSchema.groovy`)
```